sql/atxs/atxs_test.go

```diff
@@ -533,17 +530,12 @@ func TestVRFNonce(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, nonce1, got)
 
-	got, err = atxs.VRFNonce(db, sig.NodeID(), atx3.TargetEpoch())
-	require.NoError(t, err)
-	require.Equal(t, nonce3, got)
-
-	// between epochs returns previous nonce
 	got, err = atxs.VRFNonce(db, sig.NodeID(), atx2.TargetEpoch())
 	require.NoError(t, err)
-	require.Equal(t, nonce1, got)
+	require.Equal(t, nonce3, got)
```

Comment:
This changed the meaning of the test. The point was that if I have 3 ATXs:

- ATX1 with nonce
- ATX2 without nonce
- ATX3 with nonce

and I query for the targetEpoch of ATX2, I should get the nonce of ATX1. With the change I should now get the nonce of ATX2 (which is the same as the one set in ATX1). But not the one for ATX3.

Reply:
The part `ATX2 without nonce` makes no sense anymore (starting from the 17th migration) as every ATX in the DB has a VRF nonce.

The test still verifies if querying for a VRF nonce returns a nonce from the _last_ ATX. Btw, we should probably verify if this is a valid way to query a smesher's nonce in the first place as it might be ambiguous if the smesher published more than 1 ATX in the last epoch when he published an ATX.

Reply:
I will update the test to make it more clear.

Reply:
The test is till unclear for me. We have the following ATXs:

- ATX1: TargetEpoch 20, VRFNonce 333
- ATX2: TargetEpoch 30, no VRFNonce
- ATX3: TargetEpoch 50, VRFNonce 777

Then we asserted for the identity that created those ATXs that it got the following VRFNonces for the following target epochs:

- Epoch 20: 333
- Epoch 30: 333
- Epoch 50: 777
- Epoch 60: 777

Now we only have 2 ATXs and one less assert:

- ATX1: TargetEpoch 20, VRFNonce 333
- ATX2: TargetEpoch 50, VRFNonce 777
- Epoch 20: 333
- Epoch 50: 777
- Epoch 60: 777

The - in my opinion crucial - test case for the ATX without a VRF nonce in epoch 30 is missing now.

Reply:
The situation with `ATX2: TargetEpoch 30, no VRFNonce` is impossible to come upon. Every ATX in the `atxs` table has a nonce.

Reply:
OK, then just the naming is off with `nonce3` 😅